include/deal.II/numerics/fe_field_convolution_function.h

```diff
@@ -0,0 +1,280 @@
+#include <deal.II/base/function_lib.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/mapping_q1.h>
+
+#include <deal.II/grid/grid_tools_cache.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/meshworker/scratch_data.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+  /**
+   * This is an approximated interpolation function for the given dof handler
+   * and the given solution vector, that actually performs a convolution with
+   * a specified kernel in order to compute its values.
+   *
+   * In particular, the evaluation of this function at a point $x$ is computed
+   * as:
+   *
+   * \f[
+   * F(x) = \int_{\Omega} f(y) K^{\epsilon}(x-y) dy
+   * \f]
+   * where $f(y)$ is the finite element field function constructed using the
+   * vector and the dof handler passed at construction time.
+   *
+   * This function takes a CutOffFunctionBase object in input, that is used to
+   * perform the convolution, and a radius, that is used to scale the
+   * convolution kernel. This function may be called also for points
+   * outside the domain, where it decays to zero according to what cut off
+   * function you used to construct this object.
+   *
+   * An example usage is the following:
+   * @code
+   *
+   * // Generate two triangulations
+   * Triangulation<dim> tria_1;
+   * Triangulation<dim> tria_2;
+   *
+   * // Read the triangulations from files, or build them up, or get them
+   * // from some place.
+   * ...
+   *
+   * // Associate a dof handler and a solution to the first triangulation
+   * DoFHandler<dim> dh1 (tria_1);
+   * Vector<double> solution_1;
+   *
+   * // Do the same with the second
+   * DoFHandler<dim> dh2 (tria_2);
+   * Vector<double> solution_2;
+   *
+   * // Setup the system, assemble matrices, solve problems and get the
+   * // nobel prize on the first domain...
+   * ...
+   *
+   * // Create a convolution kernel, and a convolution radius
+   * Functions::CutOffFunctionC1<dim> kernel;
+   * const double radius = 2*h;
+   *
+   * // Now build the convolution of the first function, projected on the
+   * // second domain
+   * FEFieldConvolutionFunction<dim> fe_function_1 (dh_1, solution_1,
+   *                                                kernel, radius);
+   * VectorTools::project (dh_2, constraints_2, quad,
+   *                       fe_function_1, solution_2);
+   *
+   * // Or interpolated...
+   * Vector<double> solution_3;
+   * VectorTools::interpolate (dh_2, fe_function_1, solution_3);
+   *
+   * @endcode
+   *
+   * FEFieldConvolutionFunction is designed to be an easy way to get the results
+   * of your computations across different, possibly non matching, grids. No
+   * knowledge of the location of the points is assumed in this class.
+   *
+   * <h3>Using FEFieldConvolutionFunction with
+   * parallel::distributed::Triangulation</h3>
```

Comment:
I think it has to be in the same line

Reply:
clang-format refuses to comply. I'll take a look at the generated documentation to see if this is an issue.